tool/tsh/kube.go

```diff
@@ -604,6 +647,12 @@ func (c *kubeCredentialsCommand) run(cf *CLIConf) error {
 		}
 	}
 
+	unlockKubeCred, err := takeKubeCredLock(cf.Context, cf.HomePath, cf.Proxy)
```

Comment:
I don't think we really need it.

Reply:
I think we do: cluster `foo` can be down, and cluster `bar` be up and running. You shouldn't block access to a running cluster just because the user tried to access a cluster which is down before it. There is no way to achieve that behavior if you have just a single lock.

This argument goes for both Teleport and k8 clusters.

Reply:
Yep, I mean we don't really need in practice, even though there can be some variations in theory. Realistically what we're up against is expired login of the user's SSO provider, and it's unlikely for a user to have multiple SSO providers per teleport cluster on the same proxy. As I mentioned above, I don't think we should overcomplicate the solution, and this doesn't block user's manual login, only automated cert reissuing.

Reply:
All right, we can always make the locks more granular in the future.